index.js

```diff
@@ -44,49 +34,126 @@ app.get('/docs', (req, res) => {
   res.send(apiDocs);
 });
 
-app.get('/status', (req, res) => {
-  res.send({
-    'statusCode': wallet.state,
-    'statusMessage': humanState[wallet.state],
-    'network': wallet.network,
-    'serverUrl': wallet.server,
-    'serverInfo': wallet.serverInfo,
+app.post('/start', (req, res) => {
+  // We expect the user to send the seed he wants to use
+  // if not sent, we use the 'default' key
+  const seedKey = req.body.seedKey || 'default';
+  if (!(seedKey in config.seeds)) {
+    res.send({
+      success: false,
+      message: 'Seed not found.',
+    });
+    return;
+  }
+
+  // The user must send a key to index this wallet
+  if (!('key' in req.body)) {
+    res.send({
+      success: false,
+      message: 'Parameter \'key\' is required.',
+    });
+    return;
+  }
+
+  const seed = config.seeds[seedKey];
+
+  const walletConfig = {
+    network: config.network,
+    server: config.server,
+    seed,
+  }
+
+  // tokenUid is optionat but if not passed as parameter
+  // the wallet will use HTR
+  if (config.tokenUid) {
+    walletConfig['tokenUid'] = config.tokenUid;
+  }
+
+  // Passphrase is optional but if not passed as parameter
+  // the wallet will use empty string
+  if (req.body.passphrase) {
+    walletConfig['passphrase'] = req.body.passphrase;
+  }
+
+  const wallet = new HathorWallet(walletConfig);
+  wallets[req.body.key] = wallet;
```

Comment:
Should it only be added after `wallet.start` succeeds, inside the promise?

Reply:
Yeah, it's better